src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAssertion.cs

```diff
@@ -152,6 +157,46 @@ public string MinorVersion
         /// </summary>
         public Signature Signature { get; set; }
 
+        /// <summary>
+        /// Gets the canonicalized (ExclusiveC14n) representation without comments.
+        /// </summary>
+        public string CanonicalString
```

Comment:
Would it make sense to convert this property to a method, allowing users to specify includeComments and inclusiveNamespacesPrefixList arguments?

Reply:
Probably not, the comment is a bit misleading.

If the Assertion was created by the EnvelopedSignatureReader, the XmlTokenStream has the correct nodes. However, i don't believe we are remembering the 'includeComments and inclusiveNamespacesPrefixList when we capture the XmlTokenStream. I believe we may need to do that.

Reply:
We decided to make this internal for now. It is only available when the EnvelopedSignatureReader is involved. We pass the canonicalized string for replay detection and for the bootstrap token for OBO flows.